src/Peering/Peering.Test/ScenarioTests/PeeringServiceTests.ps1

```diff
@@ -0,0 +1,138 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+<#
+.SYNOPSIS
+GetPeeringServiceProviders 
+#>
+function Test-GetPeeringServiceProviders {
+    #static names are used because providers are not random.
+    $name = "TestPeer1"
+    $provider = Get-AzPeeringServiceProvider
+    Assert-NotNull $provider
+    Assert-AreEqual $name $provider[0].ServiceProviderName
+}
+
+<#
+.SYNOPSIS
+GetPeeringServiceLocations
+#>
+function Test-GetPeeringServiceLocations {
+    $locations = Get-AzPeeringServiceLocation -Country "United States"
+    Assert-NotNull $locations
+    $state = $locations | Where-Object { $_.Name -match "Washington" }
+    Assert-NotNull "Washington" $state
+    $locations = Get-AzPeeringServiceLocation -Country "United States"
+    Assert-NotNull $locations
+}
+
+<#
+.SYNOPSIS
+GetPeeringServiceProviders 
+#>
+function Test-GetPeeringServiceByResourceGroup {
+    #Hard Coded locations becuase of limitations in locations
+    $name = getAssetName "myPeeringService";
+    $loc = "Washington"
+    $provider = "TestPeer1"
+    $resourceGroup = "Building40"
+    $peeringService = New-AzPeeringService -ResourceGroupName $resourceGroup -Name $name -Location $loc -PeeringServiceProvider $provider
+    Assert-NotNull $peeringService
+    Assert-AreEqual $peeringService.Name $name
+    Assert-AreEqual $loc $peeringService.PeeringServiceLocation
+    Assert-AreEqual $provider $peeringService.PeeringServiceProvider
+    $peeringService = Get-AzPeeringService -ResourceGroupName $resourceGroup -Name $name
+    Assert-NotNull $peeringService
+    Assert-AreEqual $peeringService.Name $name
+    Assert-AreEqual $loc $peeringService.PeeringServiceLocation
+    Assert-AreEqual $provider $peeringService.PeeringServiceProvider
+}
+
+<#
+.SYNOPSIS
+GetPeeringServiceLocations
+#>
+function Test-GetPeeringServiceByResourceId {
+    #Hard Coded locations becuase of limitations in locations
```

Comment:
@rileymckenna we should make sure that these tests delete the resources created during the test. Usually the pattern is the following:

```powershell
# Create resource group
try
{
    # Create resources needed for test in resource group

    # Validate cmdlets
}
finally
{
    # Delete resource group
}
```

Reply:
We dont support delete for peering or peering service because those cause us to have to go onto a router and remove the sessions. This can be risky so right now, we only support creates and updates. In the future we will do a new cmdlet review to add Remove-AzPeering and Remove-AzPeeringService.